### PR TITLE
Add missing functionalities to kani_core

### PR DESCRIPTION
**Pull request by celinval**
- Add Arbitrary for array
  - Add Arbitrary for tuples
  - Add missing changes from modifies slices (#3295)

Note that for adding `any_array` I had to cleanup the unnecessary usage of constant parameters from `kani::any_raw`.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: library/kani/src/lib.rs [hunk 1]
```diff
@@ -7,11 +7,10 @@
 // Used for rustc_diagnostic_item.
 // Note: We could use a kanitool attribute instead.
 #![feature(rustc_attrs)]
-// This is required for the optimized version of `any_array()`
-#![feature(generic_const_exprs)]
-#![allow(incomplete_features)]
 // Used to model simd.
 #![feature(repr_simd)]
+#![feature(generic_const_exprs)]
```
Comment (jaisnan) [inline library/kani/src/lib.rs:12]: Is this a clippy cleanup?

**Review by jaisnan — APPROVED** [review]
Thanks for the cleanup, this might make unification easier as well. Approving as the added function-contracts in the `verify_core` seem to check for regressions in newly added capabilities as well.

File: library/kani/src/lib.rs (hunk 1 above)
Comment (celinval) [reply library/kani/src/lib.rs:12]: No. I updated it because before it was needed for `simd` and any_array. However, now it is only needed for `simd`.

**Comment by celinval** [comment]
> Thanks for the cleanup, this might make unification easier as well. Approving as the added function-contracts in the `verify_core` seem to check for regressions in newly added capabilities as well.

BTW, I noticed that `core` crate contains a re-export of itself as `core`, so they can use `core` inside it. So we might be able to just use straight `core::` for both versions (with and without `core`), so we may not need the `$core` argument to our macros.

**Pull request merged**